### DISCUSSION TITLE
Error with reverse proxy HTTP/HTTPS

### DISCUSSION THREAD

**Discussion by Flappes** [snipe/snipe-it#13405] [General]
Hi,

i've got a running local installation of snipe-it, version 6.12, using http. PHP=7.4.33, Laravel=8.83.22, Debian11.
i tried to put a reverse proxy in front of the installation (a separate utm-firewall), which i've done succesfully with any other of our local server applications. My first GET-request is translated to https, but following request still used http. After a little try'n'error i come across the setting APP_FORCE_TLS which i set to true, then issuing  _php artisan config:clear_. It works instantly. Yes!!
But during fixing my other issue with laravel not logging anything in /storage/logs/laravel.log i tried to fix a problem with _SocialiteServiceProvider which could not be found_, which i reinstalled with _composer require laravel/socialite_ and _php artisan config:cache_. 
After this change only the first request again is https, any following are http. It works again when issuing _php artisan config:clear_.
**So in short**: when i use php artisan config:clear everything seems to be ok. when i use php artisan config:cache only the first request is https. Then i cannot login because of browser security i think.
Anyone can help me with this?

**Comment by snipe** [comment]
You should not be editing (via composer update) the composer.lock. When you do so, we have no way of knowing what changed within the package, and unexpected errors occur.

What's your APP_URL?

**Reply by Flappes** [reply]
I had changed it from **http**://snipeit.domain.com to **https**://snipeit.domain.com

**Comment by Flappes** [comment]
I have restored the vm snapshot to version 6.10, runnning with http. upgraded again to 6.12 with chown _user_ and chmod 775. Then switched to _user_ and upgraded with _php upgrade.php_. Laravel.log shows again the socialite Errormessage:
[2023-08-04 09:34:00] production.ERROR: Class 'Laravel\Socialite\SocialiteServiceProvider' not found {"exception":"[object] (Error(code: 0): Class 'Laravel\\Socialite\\SocialiteServiceProvider' not found at /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/ProviderRepository.php:208).
I have enabled debug-mode in .env, but no entries in /storage/logs/laravel.log. I don't use login providers provided by Socialite, so it wouldn't bother me if the class is missing. But I would like to know why nothing is logged in Laravel. Can someone tell me how to provoke an entry in laravel.log to make sure that logging works or doesn't work?
I would then postpone the conversion to https until later

**Reply by Flappes** [reply]
No idea from anyone?
